binary_search_tree/tree.py

```diff
@@ -1,59 +1,114 @@
 class TreeNode:
-    def __init__(self, key, val = None):
+    def __init__(self, key, val=None):
         if val == None:
             val = key
 
         self.key = key
         self.value = val
         self.left = None
         self.right = None
-        
-
 
 class Tree:
     def __init__(self):
         self.root = None
-
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
+    def add_helper(self, current, key, value):
```

Comment:
👍 Since this is recursive the space complexity is O(log n) assuming the tree is balanced and O(n) otherwise.